packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts

```diff
@@ -196,16 +201,19 @@ export class NetworkLoadBalancer extends BaseLoadBalancer implements INetworkLoa
   public readonly metrics: INetworkLoadBalancerMetrics;
   public readonly securityGroups?: string[];
   public readonly ipAddressType?: IpAddressType;
+  public readonly connections: ec2.Connections;
 
   constructor(scope: Construct, id: string, props: NetworkLoadBalancerProps) {
     super(scope, id, props, {
       type: 'network',
-      securityGroups: props.securityGroups?.map(sg => sg.securityGroupId),
+      securityGroups: Lazy.list({
+        produce: () => this.connections.securityGroups.length >= 1 ? this.connections.securityGroups.map(sg => sg.securityGroupId) : undefined,
+      }),
```

Comment:
If `securityGroups` becomes an empty array from undefined, an update will be applied and deployment will not be possible, so in the case of an empty array, it is undefined for backwards compatibility.